credentials.go

```diff
@@ -197,21 +263,38 @@ func (c *CredentialManager) Create(timestamp time.Time, nodeID []byte, OperatorT
 }
 
 // Verify checks that a AuthenticatedCredential has a valid mac
-func (c *CredentialManager) Verify(authenticatedCredential *AuthenticatedCredential) error {
+func (c *CredentialManager) Verify(authenticatedCredential *AuthenticatedCredential) (*ID, error) {
 	// Create a temporary AuthenticatedCredential and borrow the inner message from the provided credential
 	tmp := AuthenticatedCredential{}
-	tmp.Credential = authenticatedCredential.Credential
-
-	// Auth tmp
-	if err := c.authenticateCredential(&tmp); err != nil {
-		return errors.Wrap(err, "Error while re-creating the MAC")
+	tmp.Credential = &pb.Credential{}
+	proto.Merge(tmp.Credential, authenticatedCredential.Credential)
+	bytes, err := proto.Marshal(tmp.Credential)
```

Comment:
Do we need the `tmp` variable? Could we use `proto.Marshal(authenticatedCredential.Credential)` instead?